cmd/api-mock/main.go

```diff
@@ -342,3 +366,29 @@ func unmarshalPayloadToAddonStatus(data []byte) (addonStatus, error) {
 	}
 	return status, nil
 }
+
+type APIMockConfig struct {
+	// Use string to simplify checking of values that weren't set
+	FailOnAddonStatusCreateEndpoint string `json:"failOnAddonStatusCreateEndpoint"`
+}
+
+var apiMockConfig = APIMockConfig{
+	FailOnAddonStatusCreateEndpoint: "false",
+}
+
+// Configures this api mock server on-the-fly
+func configure(w http.ResponseWriter, r *http.Request) {
+	decoder := json.NewDecoder(r.Body)
+
+	var conf APIMockConfig
+
+	if err := decoder.Decode(&conf); err != nil {
+		fmt.Printf("Failed to decode config: %v\n", err)
+		w.WriteHeader(http.StatusBadRequest)
+	} else {
+		fmt.Printf("Patching server with: %v\n", conf)
+		if len(conf.FailOnAddonStatusCreateEndpoint) > 0 {
```

Comment:
can be better handled with reflect.TypeOf(conf.FailOnAddonStatusCreateEndpoint) is bool then the assignment